src/Servers/Kestrel/Transport.MsQuic/MsQuicTransportOptions.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.MsQuic
+{
+    public class MsQuicTransportOptions
+    {
+        public ushort MaxBidirectionalStreamCount { get; set; } = 100;
+        public ushort MaxUnidirectionalStreamCount { get; set; } = 10;
+        public string Alpn { get; set; } = "h3-23";
+        public string RegistrationName { get; set; } = "Quic";
```

Comment:
Don't set non-zero defaults. Require the app to set an ALPN. And for the registration name at least default to something like asp.net so we can differentiate different code in the same process.

Reply:
This was done due to a layering issue with Kestrel we need to figure out. Ideally, if Kestrel is configured to use HTTP/3, it itself will set the alpn and registration name. I'll remove them though and put them in the app as you are correct.